pkg/auth/authorizer/abac/abac.go

```diff
@@ -100,13 +101,19 @@ func NewFromFile(path string) (policyList, error) {
 func (p policy) matches(a authorizer.Attributes) bool {
 	if p.subjectMatches(a) {
 		if p.Readonly == false || (p.Readonly == a.IsReadOnly()) {
-			if p.Resource == "" || (p.Resource == a.GetResource()) {
+			switch {
+			case p.NonResourcePath != "":
+				if p.NonResourcePath == a.GetNonResourcePath() {
+					return true
+				}
+			case p.Resource == "" || (p.Resource == a.GetResource()):
```

Comment:
If `len(a.GetNonResourcePath()) != 0`, then this branch of the switch should never return true.